### ISSUE TITLE
`redundant_as_str` misses chained method calls

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#16979]
### Summary

The lint currently does not trigger for `.as_str().len()` even though the `.as_str()` call is unnecessary.

Reproducer: https://rust.godbolt.org/z/5s563YbP3

### Lint Name

`redundant_as_str`

### Reproducer

I tried this code:

```rust
fn issue16979(s: String) -> usize {
    s.as_str().len()
}
```

I expected to see this happen:

```text
warning: this `as_str` is redundant and can be removed as the method immediately following exists on `String` too
 --> <source>:5:7
  |
5 |     s.as_str().len()
  |       ^^^^^^^^^^^^^^^^^ help: try: `len`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_as_str
```

Or something similar indicating that the `.as_str()` call can be removed.

Instead, this happened:

No lint ouput

### Version

```text
clippy 0.1.97 (fb0a5a5a9c 2026-05-08)
```

**Comment by profetia** [comment]
The existing implementation hardcoded `.is_empty()` and `.as_bytes()`. It do worth some improvements.
